### ISSUE TITLE
Does it work on Raspbian Buster?

### ISSUE THREAD

**Issue by karanda1990** [eXist-db/exist#3168]
Hi folks, 

I wanted to test-install on a RPi 3, since the issue with the user-manager prevents me from developing on my server. The Pi is only accessible from my private network! 

However, after reading [this](http://exist-db.org/exist/apps/doc/basic-installation), atlhough it started after running the command from terminal, I could not go to localhost:8080 without changing anything in terms of standard-places for DBs etc. 
Neither could I cd to tools/yajsw and run any script, since there was no such directory, according Raspbian.

Any hints? 
best wishes

EDIT: some screenshots from installation process. attached! 
![2019-12-17-194805_1920x1080_scrot](https://user-images.githubusercontent.com/58807751/71025639-be55c080-2107-11ea-8fc8-de220d51254e.png)
![2019-12-17-194906_1920x1080_scrot](https://user-images.githubusercontent.com/58807751/71025641-beee5700-2107-11ea-8dc6-8e2f7d06e396.png)
![2019-12-17-193943_1920x1080_scrot](https://user-images.githubusercontent.com/58807751/71025642-beee5700-2107-11ea-87f7-540acc8f88e0.png)

K

**Comment by joewiz** [comment]
There shouldn't be any problem running eXist on any system that meets the [System Requirements](http://exist-db.org/exist/apps/doc/basic-installation#system-requirements). And from your screenshots, I don't see any problems. There are warnings, but not errors. Did you finish installing eXist? Does eXist start? eXist no longer ships with yajsw. Please help us understand what problems and questions you have.

**Comment by karanda1990** [comment]
I'd really love to hellp you further... 
I did install it following the basic installation steps step by step. As I did on the webserver, before I moved to docker. 

I installed it and when I launched it, I was asked the "database stuff", such as max. amount of storage/RAM etc. I again did not change the standards, entered a password and hit next.
Nothing happened and ps -e did not show a running exist-db process. So I started exist-db from the starter menu - to no avail. I checked with ps -e whether it is running. 
I am using the standard-jdk that raspbian comes with (Open-JDK 11). 
Unfortunately I cannot remember which JDK was running on the ubuntu/debian machines and the win10 is obviously oracle's implementation. (I do not have these machines around, I lent them to friends, as I only use them for testing and "playing")

I fear that is not too helpful and by far no way of asking for help, but the many times before (on local machine with ubuntu, win10, debian it just worked.

**Comment by dizzzz** [comment]
hmmm isn't Rasbian a 32bits linux with a 32bits java? that is not sufficient to run eXist-db on it.

please could you check `java --version` ?

**Comment by dizzzz** [comment]
answering myself: raspberrypi is 32bits (on 64bits hardware). That is not sufficient for eXist-db default memory settings.

```
pi@raspberrypi:~ $ java -version
openjdk version "11.0.5" 2019-10-15
OpenJDK Runtime Environment (build 11.0.5+10-post-Raspbian-1deb10u1)
OpenJDK Server VM (build 11.0.5+10-post-Raspbian-1deb10u1, mixed mode)
```

on mac:

```
 java -version
openjdk version "11.0.5" 2019-10-15 LTS
OpenJDK Runtime Environment Zulu11.35+15-CA (build 11.0.5+10-LTS)
OpenJDK 64-Bit Server VM Zulu11.35+15-CA (build 11.0.5+10-LTS, mixed mode)
```

**Comment by joewiz** [comment]
@dizzzz According to https://exist-db.org/exist/apps/doc/basic-installation#system-requirements, the minimum memory required is 512 MB. Is that not possible with a 32 bit system? Or do we need to update the system requirements statement?

**Comment by dizzzz** [comment]
@joewiz hmmmm indeed. When running `bin/startup.sh' I see the following heap size of 4g, but there is no `-Xmx` parameter set it looks like....

<img width="704" alt="image" src="https://user-images.githubusercontent.com/1700062/71124100-6e98f700-21e4-11ea-9fba-3584b10c36a9.png">

**Comment by dizzzz** [comment]
@karanda1990  when re-reading your post multiple times, I think  something else is wrong
==> please always fill in the complete issue template <==
`tools/yajsw` has been removed with version 5.x
your exist-db looks to be in `/usr/local/eXist-db`

so assuming you did not change anything in the default configuration settings [you did not fill in the template == essential details are missing] please type
- cd /usr/local/eXist-db
- ./bin/startup.sh

**Comment by dizzzz** [comment]
@joewiz I'd vore for almost blindly close tickets when the template is not filled in. it is most of the time not time efficient for us.

**Comment by karanda1990** [comment]
```
sudo ./bin/startup.sh
19 Dez. 2019 17:00:32,582 [main] INFO  (JettyStart.java [run]:173) - Running with Java 11.0.3 [Raspbian (OpenJDK Server VM) in /usr/lib/jvm/java-11-openjdk-armhf]
19 Dez. 2019 17:00:32,585 [main] INFO  (JettyStart.java [run]:180) - Running as user 'root'
19 Dez. 2019 17:00:32,587 [main] INFO  (JettyStart.java [run]:181) - [eXist Home : /usr/local/eXist-db]
19 Dez. 2019 17:00:32,601 [main] INFO  (JettyStart.java [run]:182) - [eXist Version : 5.1.1]
19 Dez. 2019 17:00:32,604 [main] INFO  (JettyStart.java [run]:183) - [eXist Build : 20191206200820]
19 Dez. 2019 17:00:32,606 [main] INFO  (JettyStart.java [run]:184) - [Git commit : f5b62e21e915797bed87e08846a595b4eaee14aa]
19 Dez. 2019 17:00:32,608 [main] INFO  (JettyStart.java [run]:186) - [Operating System : Linux 4.19.75-v7+ arm]
19 Dez. 2019 17:00:32,610 [main] INFO  (JettyStart.java [run]:187) - [log4j.configurationFile : /usr/local/eXist-db/etc/log4j2.xml]
19 Dez. 2019 17:00:32,791 [main] INFO  (JettyStart.java [run]:188) - [jetty Version: 9.4.24.v20191120]
19 Dez. 2019 17:00:32,793 [main] INFO  (JettyStart.java [run]:189) - [jetty.home : /usr/local/eXist-db]
19 Dez. 2019 17:00:32,795 [main] INFO  (JettyStart.java [run]:190) - [jetty.base : /usr/local/eXist-db]
19 Dez. 2019 17:00:32,797 [main] INFO  (JettyStart.java [run]:191) - [jetty configuration : /usr/local/eXist-db/etc/jetty/standard.enabled-jetty-configs]
19 Dez. 2019 17:00:35,941 [main] INFO  (JettyStart.java [run]:200) - Configuring eXist from /usr/local/eXist-db/etc/conf.xml

```

we're seemingly getting somewhere here. However, now we get the warnings from the screenshot back and it stops ...

> > WARNING: An illegal reflective access operation has occurred
> WARNING: Illegal reflective access by org.exist.storage.BrokerPool (file:/usr/local/eXist-db/lib/exist-core-5.1.1.jar) to field java.lang.Thread.threadLocals
> WARNING: Please consider reporting this to the maintainers of org.exist.storage.BrokerPool
> WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
> WARNING: All illegal access operations will be denied in a future release

How can I move on now? any more hints?

**Comment by adamretter** [comment]
> WARNING: An illegal reflective access operation has occurred

These warnings can be safely ignored. There are just warning that reflection may change in a newer JDK, but for now it is fine.

This is my systemd config. Note that I have setup a service user and group called 'existdb` to run it under. If you don't want to do that you should modify the `user` and `group` entries.

```
[Unit]
Description=eXist-db NoSQL Database
After=syslog.target

[Service]
Type=simple
User=existdb
Group=existdb
ExecStart=/usr/local/exist-db/bin/startup.sh
Restart=on-abort

[Install]
WantedBy=multi-user.target
```

**Comment by dizzzz** [comment]
still @karanda1990  the template asks for some more details; please provide.

**Comment by dizzzz** [comment]
ok `sudo ./bin/startup.sh ` so problem is solved ? the rest looks like permission issues / linux specific thingies. Probably not for this support board....

**Comment by karanda1990** [comment]
> 
> 
> ok `sudo ./bin/startup.sh ` so problem is solved ? the rest looks like permission issues / linux specific thingies. Probably not for this support board....

I do not know, why the second time `sudo ./bin/startup.sh` worked. And it does not work every time I start the service. 
Then I get the above error.

I am quite sure that I should make a user and group for running exist over the internet ... I just wanted a stable environment set up, before doing this ... :-D 

I started this thread not as a problem but as sth. else by mistake, so I never saw the template in question ... Furthermore, all the Info I was ABLE to gather or I could get with my (unargueably not very deep knowledge of exist-db and linux), I shared ... Instead of pounding on "providing" I would rather assume that the people asking are not aware of more information. 

(I get that it is frustrating to find posts where not everything is mentioned, trust me ... Been there on android-dev boards and javascript-discussions, but most of the time it is not a sign of disrespect but a sign of missing knowledge) (maybe I am reading you wrong, but it reads quite blunt)
